src/lib/swapper/swappers/CowSwapper/getCowSwapTradeQuote/getCowSwapTradeQuote.ts

```diff
@@ -92,43 +94,49 @@ export async function getCowSwapTradeQuote(
       from: DEFAULT_ADDRESS,
       kind: ORDER_KIND_SELL,
       sellAmountBeforeFee: normalizedSellAmountCryptoBaseUnit,
-    } as CowSwapSellQuoteApiInput)
+    } as CowSwapSellQuoteApiInput,
+  )
+
+  if (maybeQuoteResponse.isErr()) return Err(maybeQuoteResponse.unwrapErr())
 
-    const {
+  const {
+    data: {
       quote: {
         buyAmount: buyAmountCryptoBaseUnit,
         sellAmount: sellAmountCryptoBaseUnit,
         feeAmount: feeAmountInSellTokenCryptoBaseUnit,
       },
-    } = data
-
-    const quoteSellAmountPlusFeesCryptoBaseUnit = bnOrZero(sellAmountCryptoBaseUnit).plus(
-      feeAmountInSellTokenCryptoBaseUnit,
-    )
-
-    const buyCryptoAmount = bn(buyAmountCryptoBaseUnit).div(
-      bn(10).exponentiatedBy(buyAsset.precision),
-    )
-    const sellCryptoAmount = bn(sellAmountCryptoBaseUnit).div(
-      bn(10).exponentiatedBy(sellAsset.precision),
-    )
-    const rate = buyCryptoAmount.div(sellCryptoAmount).toString()
-
-    const approveData = getApproveContractData({
-      web3,
-      spenderAddress: COW_SWAP_VAULT_RELAYER_ADDRESS,
-      contractAddress: sellAssetErc20Address,
-    })
-
-    const [feeData, sellAssetUsdRate] = await Promise.all([
-      adapter.getFeeData({
-        to: sellAssetErc20Address,
-        value: '0',
-        chainSpecific: { from: receiveAddress, contractData: approveData },
-      }),
-      getUsdRate(deps, sellAsset),
-    ])
-
+    },
+  } = maybeQuoteResponse.unwrap()
+
+  const quoteSellAmountPlusFeesCryptoBaseUnit = bnOrZero(sellAmountCryptoBaseUnit).plus(
+    feeAmountInSellTokenCryptoBaseUnit,
+  )
+
+  const buyCryptoAmount = bn(buyAmountCryptoBaseUnit).div(
+    bn(10).exponentiatedBy(buyAsset.precision),
+  )
+  const sellCryptoAmount = bn(sellAmountCryptoBaseUnit).div(
+    bn(10).exponentiatedBy(sellAsset.precision),
+  )
+  const rate = buyCryptoAmount.div(sellCryptoAmount).toString()
+
+  const approveData = getApproveContractData({
+    web3,
+    spenderAddress: COW_SWAP_VAULT_RELAYER_ADDRESS,
+    contractAddress: sellAssetErc20Address,
+  })
+
+  const [feeData, maybeSellAssetUsdRate] = await Promise.all([
+    adapter.getFeeData({
+      to: sellAssetErc20Address,
+      value: '0',
+      chainSpecific: { from: receiveAddress, contractData: approveData },
+    }),
+    getUsdRate(deps, sellAsset),
+  ])
+
+  return maybeSellAssetUsdRate.map(sellAssetUsdRate => {
```

Comment:
Unchecked monad.

Reply:
Fixed as part of https://github.com/shapeshift/web/commit/a45340d5415b9d8084ee8bb757678f5a9dfad3db